### ISSUE TITLE
Doubts regarding inference

### ISSUE THREAD

**Issue by ajkailash** [XixiLiu95/GEN#2]
Hi,
Thank you making the paper and code available.

Your method of computing negated generalized entropy score to detect OOD is neat. However, how would I use this score during inference? For example, imagine I have a classifier network - frozen pre-trained backbone and final classification layer trained on 10 animal classes. At inference time, I want to detect when an image sample with labels not part of training comes in as input.

I can compute the GEN score for the said sample, and also assume I have GEN scores for ID test samples. What should I do next?
Thank you for your reply in advance!

**Comment by XixiLiu95** [comment]
Hi,

Once you've acquired the test data score, it's necessary to establish a threshold for determining if a test sample belongs to ID or OOD . One possible option is to use the threshold computed from the Youden index [1]. Nevertheless, the exact threshold to be used will vary based on the user's specific case.

[1] Youden WJ. Index for rating diagnostic tests. Cancer, 1950.
